sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesCalculator.cs

```diff
@@ -0,0 +1,111 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Metrics.CSharp
+{
+    public class ExecutableLinesCalculator
+    {
+        public int CountLines(SyntaxTree tree)
+        {
+            var nodesToVisit = new Queue<SyntaxNode>(tree.GetRoot().ChildNodes());
+            var executableLineNumbers = new HashSet<int>();

```

Comment:
Might sound stupid but 2 statements on the same line will count as only one executable line, is it the expected behavior?

Reply:
Yes. This is what VS metrics do and in the spirit of what Ann's article suggests.

Reply:
@duncanp-sonar can you provide a short code sample?